synapse/config/emailconfig.py

```diff
@@ -87,7 +87,10 @@ def read_config(self, config, **kwargs):
         # if they have this set and tell them to use the updated option, while using a default
         # identity server in the process.
         self.using_identity_server_from_trusted_list = False
-        if config.get("trust_identity_server_for_password_resets", False) is True:
+        if (
+            not self.account_threepid_delegate
+            and config.get("trust_identity_server_for_password_resets", False) is True
+        ):
             # Use the first entry in self.trusted_third_party_id_servers instead
             if self.trusted_third_party_id_servers:
                 self.account_threepid_delegate = self.trusted_third_party_id_servers[0]
```

Comment:
ftr: it feels dodgy to me that `account_threepid_delegate` can be set either here or in `RegistrationConfig`. It probably needs factoring out at some point.

Reply:
The whole emailconfig needs factoring at some point. I'll try to keep this in mind for when that happens. (I'll add a todo)